### PR TITLE
#18 add: Sentry

### PR DESCRIPTION
**Pull request by MaXeraph**
## Changes
- Add Sentry + secret DSN in Doppler
- Add errorHandler as a util
- Changed all console.error to use ErrorHandler in order to report a handled event to Sentry.

## Todo
- Create a Sentry app on Official Organization and edit Secret in Doppler

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
+  dsn: env.SENTRY_SECRET,
+
+  // Set tracesSampleRate to 1.0 to capture 100%
+  // of transactions for performance monitoring.
+  // We recommend adjusting this value in production
+  tracesSampleRate: 1.0,
+});
 
 // Create a new client instance
 const client = new CustomClient({
```
Comment (umar-ahmed) [inline src/index.ts:16]: I think the client also has an `"error"` event we can listen for and sent events to Sentry

File: src/commands/check.ts [hunk 2]
```diff
@@ -51,7 +52,7 @@ export async function execute(interaction: CommandInteraction) {
     }
   } catch (error) {
     // Handle errors
-    console.error("error", error);
+    errorHandler(error as Error);
```
[the remark this answers is not in the document]
Comment (MaXeraph) [reply src/commands/check.ts:55]: Do you think its viable for us to insert Sentry reporter in every error catch then ?

File: src/index.ts (hunk 1 above)
Comment (MaXeraph) [reply src/index.ts:16]: Im not aware of this in the code, where is this exactly ?

File: src/index.ts [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
 import { CustomClient } from "./client";
+import * as Sentry from "@sentry/node";
+
+Sentry.init({
+  dsn: env.SENTRY_SECRET,
+
+  // Set tracesSampleRate to 1.0 to capture 100%
+  // of transactions for performance monitoring.
+  // We recommend adjusting this value in production
+  tracesSampleRate: 1.0,
```
[the remark this answers is not in the document]
Comment (MaXeraph) [reply src/index.ts]: https://github.com/chainpatrol/discord-bot/pull/24/commits/e3452a6fafc2bdc8a0bcc70789fba69b608b0dea#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R12

File: src/index.ts (hunk 1 above)
Comment (MaXeraph) [reply src/index.ts:16]: Got it. I located the event in context.

Comment (umar-ahmed) [reply src/index.ts:16]: I think it's like this:

```js
client.on('error', err => { ... })
```

https://discord.js.org/#/docs/discord.js/main/class/Client?scrollTo=e-error

File: src/commands/check.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (umar-ahmed) [reply src/commands/check.ts]: No, probably most devs would just forget to add that too. My ideal situation is that you'd re-throw the error or have some sort of catch-all error handler + if it's a recoverable error, you'd deal with it closer to where the error is thrown.

**Comment by umar-ahmed** [comment]
Created the Sentry Project: https://chainpatrol.sentry.io/projects/discord-bot/?project=4505083324137472&statsPeriod=14d and set it in the env vars for Doppler and Railway (we deploy the production bot there).

File: src/index.ts (hunk 1 above)
Comment (MaXeraph) [reply src/index.ts:16]: https://github.com/chainpatrol/discord-bot/pull/24/commits/5c169902515660c3400b49b402141299940194d3

File: src/commands/check.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (MaXeraph) [reply src/commands/check.ts]: I'm a bit confused. If we already have a try/catch we wouldn't just send a transaction to Sentry automatically right? We would assume that if there's a try-catch, the catch would handle it nicely. And if the catch is some bad error we would have to explicitly send a transaction then.

[the remark this answers is not in the document]
Comment (MaXeraph) [reply src/commands/check.ts]: [Covering Unhandled situations where some method was not caught correctly](https://github.com/chainpatrol/discord-bot/pull/24/commits/fc9baeec68be3ffa77dd0050210a31de95880d6f)

**Review by umar-ahmed — APPROVED** [review]

**Pull request merged**